compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -442,9 +442,13 @@ fn fatally_break_rust(tcx: TyCtxt<'_>) {
     }
 }
 
+/// `expected` here is the expected number of explicit generic arguments on the trait.
 fn has_expected_num_generic_args(tcx: TyCtxt<'_>, trait_did: DefId, expected: usize) -> bool {
     let generics = tcx.generics_of(trait_did);
-    generics.count() == expected + if generics.has_self { 1 } else { 0 }
+    generics.count()
```

Comment:
Is this a fix to an unrelated bug, or is it uncovered by one of the changes here?
Also double checking, is `host_effect_index` inherited by child generics?

Reply:
this is a fix to an unrelated bug, which was found while I was trying to make `PartialEq` a `const_trait`.



> Also double checking, is `host_effect_index` inherited by child generics?

yes